cli/azd/pkg/ux/step_context.go

```diff
@@ -0,0 +1,30 @@
+package ux
+
+// Context data that is persisted between steps within an action
+type StepContext interface {
```

Comment:
Was there something that pushed you towards and interface here vs just passing in a `map[string]any`?  It feels like the ergonomics there would be a little nicer, since go has special syntax for maps.

Reply:
A map could work but it would need to be a pointer value so i decided to wrap it in an interface and handle the pointer internally and just expose the interface.